### PR TITLE
[CELEBORN-1397] Fix compilation error on branch-0.4

### PR DESCRIPTION
**Pull request by onebox-li**
### What changes were proposed in this pull request?
Fix MasterClient construct method use in MasterClientSuiteJ.

### Why are the changes needed?
MasterClient's construct method has changed by https://github.com/apache/celeborn/pull/2281 on main. It's a feature to support authentication on branch-0.5.

https://github.com/apache/celeborn/pull/2466 's backport on branch-0.4 here caused a conflict in MasterClientSuiteJ.java:319.

### Does this PR introduce _any_ user-facing change?
No.

### How was this patch tested?
Local compile test.

### CODE REVIEW

**Review by FMX — APPROVED** [review]
LGTM. Merged into branch-0.4(v0.4.1).

**Comment by FMX** [comment]
I can see that there are UT failures, can you investigate the failures?

**Comment by SteNicholas** [comment]
@FMX, https://github.com/apache/celeborn/pull/2369 does not cherry pick, which causes the failure of CI.

**Review by SteNicholas — APPROVED** [review]
LGTM.

File: common/src/main/scala/org/apache/celeborn/common/CelebornConf.scala [hunk 1]
```diff
@@ -2917,12 +2917,12 @@ object CelebornConf extends Logging {
...
     buildConf("celeborn.worker.graceful.shutdown.recoverDbBackend")
       .categories("worker")
-      .doc("Specifies a disk-based store used in local db. LEVELDB or ROCKSDB.")
+      .doc("Specifies a disk-based store used in local db. ROCKSDB or LEVELDB (deprecated).")
       .version("0.4.0")
       .stringConf
       .transform(_.toUpperCase(Locale.ROOT))
       .checkValues(Set("LEVELDB", "ROCKSDB"))
-      .createWithDefault("LEVELDB")
+      .createWithDefault("ROCKSDB")
```
Comment (mridulm) [inline common/src/main/scala/org/apache/celeborn/common/CelebornConf.scala:2925]: Wont this not cause issues given this is a change in patch version ?

Comment (pan3793) [reply common/src/main/scala/org/apache/celeborn/common/CelebornConf.scala:2925]: why should we make such changes on a patched version?

Comment (SteNicholas) [reply common/src/main/scala/org/apache/celeborn/common/CelebornConf.scala:2925]: @mridulm, I have cherry pick #2369, #2372 to branch-0.4. The failure of CI is caused by these pull request, not caused by the default value change.
BTW, @onebox-li, could you rebase branch-0.4 again? I have adjusted the order of cherry pick.

Comment (SteNicholas) [reply common/src/main/scala/org/apache/celeborn/common/CelebornConf.scala:2925]: @pan3793, add a note in the upgrade guide to explain the situation. Meanwhile, the core reason is that leveldb cannot run on arm Linux.

Comment (mridulm) [reply common/src/main/scala/org/apache/celeborn/common/CelebornConf.scala:2925]: To clarify my query - if the default value is changing due to this PR - that will break existing 0.4 deployments: for example when they are not configuring this config, and now end up with a change in `recoverDbBackend`

Comment (SteNicholas) [reply common/src/main/scala/org/apache/celeborn/common/CelebornConf.scala]: @mridulm, I understood your point. I have mentioned above that we would add a note to explain this behavior. The users could add a config that changes to LEVELDB.

Comment (mridulm) [reply common/src/main/scala/org/apache/celeborn/common/CelebornConf.scala:2925]: If 0.4 is not working on arm currently - wouldn't it not be better for arm deployments to explicitly set it to ROCKSDB ? (in other words, not to change the default ?)

To put it differently, my main concern is failure of existing deployments ... in our case, for example, we do not set this config (though we are still in testing phase) - and we would typically expect a patch to work reasonably as is with existing config (between minor or major versions, there will be a lot more scrutiny).

Comment (SteNicholas) [reply common/src/main/scala/org/apache/celeborn/common/CelebornConf.scala:2925]: @mridulm, you are right. I have removed this cherry pick from branch-0.4. I would like to resolve [CELEBORN-1380](https://issues.apache.org/jira/browse/CELEBORN-1380) for arm.

**Review by SteNicholas — APPROVED** [review]
LGTM. cc @pan3793, @mridulm, @FMX.

File: common/src/main/scala/org/apache/celeborn/common/CelebornConf.scala (hunk 1 above)
Comment (pan3793) [reply common/src/main/scala/org/apache/celeborn/common/CelebornConf.scala]: Please correct fixed versions for CELEBORN-1380, it should not go 0.4.1

**Review by pan3793 — APPROVED** [review]

File: common/src/main/scala/org/apache/celeborn/common/CelebornConf.scala (hunk 1 above)
Comment (SteNicholas) [reply common/src/main/scala/org/apache/celeborn/common/CelebornConf.scala]: cc @FMX, WDYT?

**Comment by mridulm** [comment]
LGTM

**Comment by SteNicholas** [comment]
Merging to branch-0.4(v0.4.1).

**Comment by onebox-li** [comment]
Thanks all for review, and @SteNicholas cherry-pick necessary PRs.

**Pull request closed without merging**